### ISSUE TITLE
NotificationTapped not working in Killed State 

### ISSUE THREAD

**Issue by huzaifa-saeed** [TobiasBuchholz/Plugin.Firebase#289]
I am using NotificationTapped in App.cs after InitializeComponent();
but NotificationTapped not invoking in any states.

Issue:
When i tap on notification in background and foreground state it redirect user to specific page using OnNewIntent, but in killed state it's not working. In killed state Intent is empty it just open app but not redirect user to specific page.

**Comment by huzaifa-saeed** [comment]
Hi, I am calling `FirebaseCloudMessagingImplementation.OnNewIntent(intent)` in `MainActivity.cs` from both `OnCreate(...)` and `OnNewIntent(...)`

**Comment by huzaifa-saeed** [comment]
Any update regarding this?

**Comment by TobiasBuchholz** [comment]
No, handling the notification tap should function correctly, irrespective of the app's state. Could you please provide a small sample project, that demonstrates the issue?

**Comment by AndrzejKozik** [comment]
Hi!
I've experiencied similar issues as in this thread, so I'll continue here.

Enclosed, please find a simple modification of Tim-Coop sample (feature/adame branch with 3.0.0 plugin). It is limited to Android only, and needs adjusting google-services.json and ApplicationId in csproj and in AndroidManifest.xml
The app logs visited places; log view is updated on button click.

I ran the app on mobile device and sent test notifications directly from Firebase console.

I tried several test paths, each with different app behaviour:

1) App started -> notif. tapped -> notif. received with title/body- OK
2) App not started -> not. tapped -> app started (without usual .Net splash) -> notif. received but with empty Title/Body
3) AppStarted -> notif. appeared on device-> app killed -> not. tapped -> not. received with Title/Body
4) App not started -> 2 notif. appeared -> 1st tapped -> app started and notif. received without Title/Body -> 2nd tapped -> someting strange happened (second instance of app was almost started?)
5) Added ClearTaskOnLaunch = true, LaunchMode = LaunchMode.SingleTask in MainActivity.cs (as in https://github.com/TobiasBuchholz/Plugin.Firebase/issues/264#issuecomment-2054023017) and repeated step 4) -> now second tap opens notification in the same app instance (by OnNewIntent).

What I do wrong in my code so i get such inconsistent behaviour?
[CoopTimSample.zip](https://github.com/user-attachments/files/16543173/CoopTimSample.zip)

**Comment by goatrodeosoftware** [comment]
I'm also running into this on Android and iOS; I'm calling the OnNewIntent methods in the two places mentioned and wiring up to it in my MainPage codebehind and not having the method fire.
